generate.py

```diff
@@ -54,16 +56,25 @@ class CampaignSpec:
     4. For each impression, independently samples the video completion and viewability specified by the given distributions for them.
     """
 
-    def __init__(self, edpId, sd, nd, nImp, tr, freqDistSpec, videoCompDistSpec, viewabilityDistSpec, random_seed):
+    def __init__(
+        self, edpId, mcId, cId, sd, nd, nImp, tr, freqDistSpec, videoCompDistSpec, viewabilityDistSpec, randomObject
+    ):
         self.eventDataProviderId = edpId
+        self.measurementConsumerId = mcId
+        self.campaignId = cId
         self.num_days = nd
         self.dates = [sd + datetime.timedelta(days=x) for x in range(nd)]
         self.total_impressions = nImp
         self.total_reach = tr
-        tempFreqDist = DiscreteDist(self.normalize(freqDistSpec), random_seed)
+        self.random = randomObject
```

Comment:
Thanks for taking care of random seed (for reproducible results).
I don't follow why you use randomObject here and seed in `DiscreteDist`, but it's fine and please keep them. Just want to say, feel free to discuss with me in case the distinction between randomObject and seed becomes annoying

Reply:
ah, thanks for pointing out. I thought rv_discrete only accepted an int, but it turns out [it accepts a numpy generator](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_discrete.html) as well. Changed accordingly.